Build failed when configure with miniupnp and natmap from the latest commit code.

in debian and ubuntu

when I set configure to enable miniupnp and natpmp like the following(it works on last month's code)

`./configure --enable-pthread --enable-miniupnp --enable-natpmp --enable-cap --enable-pcap CFLAGS="-O3 -DN2N_OPTION_USE_PORTMAPPING"`


seems that the n2n-tools compile failed when link:

ar rcs libn2n.a src/aes.o src/auth.o src/cc20.o src/curve25519.o src/edge_management.o src/edge_utils.o src/edge_utils_win32.o src/header_encryption.o src/hexdump.o src/json.o src/management.o src/minilzo.o src/n2n.o src/n2n_port_mapping.o src/n2n_regex.o src/network_traffic_filter.o src/pearson.o src/random_numbers.o src/sn_management.o src/sn_selection.o src/sn_utils.o src/speck.o src/tf.o src/transform_aes.o src/transform_cc20.o src/transform_lzo.o src/transform_null.o src/transform_speck.o src/transform_tf.o src/transform_zstd.o src/tuntap_freebsd.o src/tuntap_linux.o src/tuntap_netbsd.o src/tuntap_osx.o src/wire.o
gcc -pthread -L.  src/edge.o libn2n.a  -ln2n -lpcap -lnatpmp -lminiupnpc  -lcap -o src/edge
cp src/edge edge
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include    -Wall     -c -o src/supernode.o src/supernode.c
gcc -pthread -L.  src/supernode.o libn2n.a  -ln2n -lpcap -lnatpmp -lminiupnpc  -lcap -o src/supernode
cp src/supernode supernode
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include    -Wall     -c -o src/example_edge_embed_quick_edge_init.o src/example_edge_embed_quick_edge_init.c
gcc -pthread -L.  src/example_edge_embed_quick_edge_init.o libn2n.a  -ln2n -lpcap -lnatpmp -lminiupnpc  -lcap -o src/example_edge_embed_quick_edge_init
cp src/example_edge_embed_quick_edge_init example_edge_embed_quick_edge_init
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include    -Wall     -c -o src/example_edge_embed.o src/example_edge_embed.c
gcc -pthread -L.  src/example_edge_embed.o libn2n.a  -ln2n -lpcap -lnatpmp -lminiupnpc  -lcap -o src/example_edge_embed
cp src/example_edge_embed example_edge_embed
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include    -Wall     -c -o src/example_sn_embed.o src/example_sn_embed.c
gcc -pthread -L.  src/example_sn_embed.o libn2n.a  -ln2n -lpcap -lnatpmp -lminiupnpc  -lcap -o src/example_sn_embed
cp src/example_sn_embed example_sn_embed
gzip -c edge.8 > edge.8.gz
gzip -c supernode.1 > supernode.1.gz
gzip -c n2n.7 > n2n.7.gz
make -C tools
**make[1]: Entering directory '/root/n2n-3.0/tools'**
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include -I../include -g3   -c -o n2n-benchmark.o n2n-benchmark.c
gcc -pthread -L. -L..  n2n-benchmark.o  -ln2n -o n2n-benchmark
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include -I../include -g3   -c -o n2n-keygen.o n2n-keygen.c
gcc -pthread -L. -L..  n2n-keygen.o  -ln2n -o n2n-keygen
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include -I../include -g3   -c -o n2n-route.o n2n-route.c
gcc -pthread -L. -L..  n2n-route.o  -ln2n -o n2n-route
gcc -O3 -DN2N_OPTION_USE_PORTMAPPING -I./include -I../include -g3   -c -o n2n-portfwd.o n2n-portfwd.c
gcc -pthread -L. -L..  n2n-portfwd.o  -ln2n -o n2n-portfwd
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o): in function `n2n_natpmp_initialization':
n2n_port_mapping.c:(.text+0x36): undefined reference to `initnatpmp'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x78): undefined reference to `sendpublicaddressrequest'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xf7): undefined reference to `getnatpmprequesttimeout'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x116): undefined reference to `readnatpmpresponseorretry'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x253): undefined reference to `closenatpmp'
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o): in function `n2n_UPNP_GetValidIGD':
n2n_port_mapping.c:(.text+0x2d5): undefined reference to `upnpDiscover'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x3a1): undefined reference to `UPNP_GetValidIGD'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x3b1): undefined reference to `freeUPNPDevlist'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x3ec): undefined reference to `UPNP_GetExternalIPAddress'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x423): undefined reference to `strupnperror'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x47d): undefined reference to `strupnperror'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x4a6): undefined reference to `freeUPNPDevlist'
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o): in function `n2n_natpmp_port_mapping_request':
n2n_port_mapping.c:(.text+0x506): undefined reference to `sendnewportmappingrequest'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x582): undefined reference to `getnatpmprequesttimeout'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x5a1): undefined reference to `readnatpmpresponseorretry'
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o): in function `n2n_upnp_del_port_mapping.isra.0':
n2n_port_mapping.c:(.text+0x7aa): undefined reference to `UPNP_DeletePortMapping'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x7f5): undefined reference to `UPNP_DeletePortMapping'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x806): undefined reference to `strupnperror'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x833): undefined reference to `FreeUPNPUrls'
/usr/bin/ld: n2n_port_mapping.c:(.text+0x86a): undefined reference to `strupnperror'
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o): in function `n2n_upnp_set_port_mapping.isra.0':
n2n_port_mapping.c:(.text+0xa28): undefined reference to `UPNP_AddPortMapping'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xa97): undefined reference to `UPNP_AddPortMapping'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xaad): undefined reference to `strupnperror'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xadf): undefined reference to `FreeUPNPUrls'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xb13): undefined reference to `strupnperror'
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o): in function `n2n_set_port_mapping':
n2n_port_mapping.c:(.text+0xc83): undefined reference to `closenatpmp'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xd22): undefined reference to `closenatpmp'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xd70): undefined reference to `closenatpmp'
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o): in function `n2n_del_port_mapping':
n2n_port_mapping.c:(.text+0xe4e): undefined reference to `closenatpmp'
/usr/bin/ld: n2n_port_mapping.c:(.text+0xee0): undefined reference to `closenatpmp'
/usr/bin/ld: ../libn2n.a(n2n_port_mapping.o):n2n_port_mapping.c:(.text+0xf25): more undefined references to `closenatpmp' follow
collect2: error: ld returned 1 exit status
make[1]: *** [<builtin>: n2n-portfwd] Error 1
make[1]: Leaving directory '/root/n2n-3.0/tools'
make: *** [Makefile:199: tools] Error 2

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -39,7 +39,7 @@ jobs:
 
       - name: Install libraries needed for all options turned on
         run: |
-          sudo apt-get install -y libminiupnpc-dev libnatpmp-dev libpcap-dev
+          sudo apt-get install -y libminiupnpc-dev libnatpmp-dev libpcap-dev libcap-dev
 
       - name: Run minimal test set
         run: |

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -25,6 +25,28 @@ jobs:
           ./configure
           make test
 
+  smoketest_all_opts:
+    name: Smoke test With all options turned on
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Fix Checkout
+        run: |
+          git fetch --force --tags
+
+      - name: Install libraries needed for all options turned on
+        run: |
+          sudo apt-get install -y libminiupnpc-dev libnatpmp-dev libpcap-dev
+
+      - name: Run minimal test set
+        run: |
+          ./autogen.sh
+          ./configure --enable-pthread --enable-miniupnp --enable-natpmp --enable-cap --enable-pcap CFLAGS="-O3 -DN2N_OPTION_USE_PORTMAPPING"
+          make test
+
   lint:
     name: Code syntax
     runs-on: ubuntu-latest

--- a/scripts/hack_fakeautoconf.sh
+++ b/scripts/hack_fakeautoconf.sh
@@ -8,7 +8,7 @@ CC=gcc
 AR=ar
 CFLAGS=$CFLAGS -g -O2 -I./include
 LDFLAGS=$LDFLAGS -L.
-N2N_LIBS_EXTRA=$LDLIBS
+LDLIBS_EXTRA=$LDLIBS
 EOF
 
 cat >tools/config.mak <<EOF

--- a/tools/Makefile
+++ b/tools/Makefile
@@ -11,6 +11,7 @@ DEBUG?=-g3
 HEADERS=$(wildcard include/*.h)
 CFLAGS+=-I../include
 LDLIBS+=-ln2n
+LDLIBS+=$(LDLIBS_EXTRA)
 
 ifeq ($(CONFIG_TARGET),mingw)
 CFLAGS+=-I../win32

